pennylane/devices/default_qubit.py

```diff
@@ -139,6 +139,7 @@ class DefaultQubit(QubitDevice):
         "QubitSum",
         "OrbitalRotation",
         "QFT",
+        "Barrier",
```

Comment:
If we don't include this here, then the gate will get expanded away to nothing.

Right now, `"default.qubit"` queries the `matrix` (just the identity) and then applies the identity to the statevector.  This seems like a lot of completely unnecessary work, when we can just expand it away before computation.